src/iothread.c

```diff
@@ -478,6 +611,18 @@ int processClientsFromIOThread(IOThread *t) {
             }
         }
 
+        /* Check if we need to run replication for master client.
+         * Run cron more frequently during failover - see replicationCron */
+        int repl_cron_freq = 1000;
+        if (server.failover_state != NO_FAILOVER)
```

Comment:
I missed this scenario, so maybe the IO thread should send the master client to main thread every calling IOThreadClientsCron

Reply:
i am not object your idea. maybe we can try it. fetch the master client into the main thread, then assign it to the io thread if needed.(usually the replica will become master, and the master client will be freed)

Reply:
after second thought, maybe we should adopt your idea, if so, we don't need to maintain master client in io thread, just treat it as a normal client, right?

Reply:
Yes sorry I didn't clarify enough. My idea was master client to be in IO-thread (after RDB replication is done) but if there is failover to move it back to be handled by main thread (i.e we can simply return 1 in `isClientMustBeHandledByMainThread` if there is failover and the master will be fetched to main). WDYT?